lib/promoted/ruby/client.rb

```diff
@@ -41,22 +44,23 @@ def initialize (params={})
 
           @delivery_timeout_millis = params[:delivery_timeout_millis] || DEFAULT_DELIVERY_TIMEOUT_MILLIS
           @metrics_timeout_millis  = params[:metrics_timeout_millis] || DEFAULT_METRICS_TIMEOUT_MILLIS
-          @api_key                 = params[:api_key] || nil
 
-          @http_client = AsyncHTTPClient.new
+          @http_client = FaradayHTTPClient.new
+          @pool = Concurrent::CachedThreadPool.new
         end
         
+        def close
```

Comment:
Do they have to manually call close?  Sorry, just a super basic Ruby question.

We should probably just get people to create this client as roughly a singleton.

Reply:
Yes, you have to manually call close, its purpose is to allow the pool to drain. And yes, I expect this to be a singleton on the client side, although I'd have to look for a pattern (beyond just docs) to clarify that.